app/background.ts

```diff
@@ -0,0 +1,90 @@
+/// <reference path="../node_modules/@types/chrome/chrome-app.d.ts" />
+import { RequestObj } from './request_list'
+
+interface Data {
+  tabId: number,
+  enabled: boolean,
+  requests: Array<RequestObj>
+  count: number
+};
+
+const DEFAULT_DATA : Data = {
+  tabId: -1,
+  enabled: false,
+  requests: [],
+  count: 0
+};
+
+class BackgroundWorker {
+  data : Array<Data> = [];
+
+  findItem = (tabId : number) => {
+    for (let index = 0; index < this.data.length; index++) {
+      const data = this.data[index];
+
+      if (Boolean(data) && data.tabId === tabId) {
+        return data;
+      }
+    }
```

Comment:
I'm not getting find's type annotations in TS, so did this. Will move that to `find` once I figure out that part.